src/fuzzer_client.c

```diff
@@ -0,0 +1,814 @@
+        }
+
+        /* Exponential backoff */
+        int delay = RECONNECT_BASE_DELAY_MS * (1 << attempts);
+        logMessage(LOG_INFO, "[Thread %d] Reconnection attempt %d failed, retrying in %d ms...",
+                   thread_id, attempts + 1, delay);
+        usleep(delay * 1000);
+        attempts++;
+    }
+
+    logMessage(LOG_ERROR, "[Thread %d] Failed to reconnect after %d attempts", thread_id, attempts);
+    return NULL;
+}
+
+/* Command execution result codes */
+typedef enum {
+    CMD_SUCCESS = 0,      /* Command executed successfully */
+    CMD_ERROR = 1,        /* General error occurred */
+    CMD_DISCONNECTED = 2, /* Connection closed by server */
+    CMD_TIMEOUT = 3       /* Command timed out */
+} CommandResult;
+
+static void handleMalformedReply(valkeyContext *ctx, Command *cmd) {
+    /* Protocol error (malformed reply) - ABORT ALL THREADS */
+    sds cmd_str = formatCommandString(cmd);
+    char error_msg[512];
+    snprintf(error_msg, sizeof(error_msg), "MALFORMED REPLY from server: %s", ctx->errstr);
+
+    /* Log the full error regardless of debug level */
+    logMessage(LOG_ERROR, "[Thread %d] CRITICAL ERROR - MALFORMED REPLY DETECTED", thread_id);
+    logMessage(LOG_ERROR, "[Thread %d] Error: %s", thread_id, ctx->errstr);
+    logMessage(LOG_ERROR, "[Thread %d] Command that caused malformed reply: %s", thread_id, cmd_str);
+
+    /* Print the exact malformed reply data if available */
+    if (ctx->reader && ctx->reader->buf && ctx->reader->len > 0) {
+        fprintf(stderr, "\n===== MALFORMED REPLY DATA BEGIN =====\n");
+        /* Print both hex and ASCII representation of the reply */
+        for (size_t i = 0; i < ctx->reader->len && i < 1024; i++) {
+            if (i % 16 == 0) {
+                fprintf(stderr, "\n%04zx: ", i);
+            }
+            fprintf(stderr, "%02x ", (unsigned char)ctx->reader->buf[i]);
+            if (i % 16 == 15) {
+                fprintf(stderr, "  ");
+                for (size_t j = i - 15; j <= i; j++) {
+                    char c = ctx->reader->buf[j];
+                    fprintf(stderr, "%c", (c >= 32 && c <= 126) ? c : '.');
+                }
+            }
+        }
+        fprintf(stderr, "\n===== MALFORMED REPLY DATA END =====\n");
+    } else {
+        fprintf(stderr, "\nNo malformed reply data available in context\n");
+    }
+
+    logMessage(LOG_ERROR, "[Thread %d] ABORTING ALL THREADS due to malformed reply (pthread_id: %lu)",
+               thread_id, (unsigned long)pthread_self());
+
+    /* Set global abort flag to stop all threads */
+    global_abort_all_threads = 1;
+
+    incrCounter(COUNTER_MALFORMED_REPLIES);
+    sdsfree(cmd_str);
+}
+
+static void handleCommandTimeout(Command *cmd, time_t start_time) {
+    /* Timeout error */
+    time_t current_time = time(NULL);
+    double elapsed_time = difftime(current_time, start_time);
+
+    char cmd_buffer[256];
+    formatCommandToBuffer(cmd, cmd_buffer, sizeof(cmd_buffer));
```

Comment:
Might want to use sds. Should we truncate or just log the entire command ?

Reply:
Thanks, I changed it to use the printCommand function. I limit the size to 1K since some args can be very big.